firefox-ios/Client/Frontend/Browser/TabScrollController/ToolbarViewProtocol.swift

```diff
@@ -0,0 +1,173 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import SnapKit
+
+protocol ToolbarViewProtocol: AnyObject {
+    var header: BaseAlphaStackView { get }
+    var topBlurView: UIVisualEffectView { get }
+    var bottomContainer: BaseAlphaStackView { get }
+    var bottomBlurView: UIVisualEffectView { get }
+    var overKeyboardContainer: BaseAlphaStackView { get }
+    var isBottomSearchBar: Bool { get }
+    var headerTopConstraint: Constraint? { get }
+    var bottomContainerConstraint: Constraint? { get }
+    var overKeyboardContainerConstraint: Constraint? { get }
+}
+
+protocol ToolbarAnimatorDelegate: AnyObject {
```

Comment:
nit: We can add the delegate directly in the `ToolbarAnimator` class like we did in `TabScrollHandler` if we want to keep working with the same pattern.